app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -288,19 +321,29 @@ private int getStatusBarHeight() {
      * Starts the PokeFly background service which contains overlay logic.
      */
     private void startPokeFly() {
-        ((Button) findViewById(R.id.start)).setText(R.string.main_stop);
+        pokeFlyStarting = true;
+        updateLaunchButtonText(true); //Should be subsumed by onResume later, which seems to be always called in
```

Comment:
4. Don't call updateLaunchButton() here.
5. Disable the button instead to avoid further touch events.